cirq-core/cirq/ops/pauli_string.py

```diff
@@ -1159,11 +1159,49 @@ def __mul__(self, other):
             return self._as_pauli_string() * other._as_pauli_string()
         if isinstance(other, (PauliString, complex, float, int)):
             return self._as_pauli_string() * other
+        if isinstance(other, gate_operation.GateOperation):
```

Comment:
group this logic into a private method (e.g. `def _try_interpret_as_pauli_string(op: Any)`) and use it in both places